src/Adapter/PlatformServices.Desktop/Services/DesktopTestSourceHost.cs

```diff
@@ -263,6 +179,129 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Updates child-domain's appbase to point to test source location and sets up
+        /// Assembly resolver for both parent and child appdomain
+        /// </summary>
+        public void UpdateAppBaseToTestSourceLocationAndSetupAssemblyResolver()
+        {
+            List<string> resolutionPaths = this.GetResolutionPaths(this.sourceFileName, VSInstallationUtilities.IsCurrentProcessRunningInPortableMode());
+
+            // Check if user specified any runsettings
+            MSTestAdapterSettings adapterSettings = MSTestSettingsProvider.Settings;
+
+            if (resolutionPaths != null && resolutionPaths.Count > 0)
+            {
+                if (EqtTrace.IsInfoEnabled)
+                {
+                    EqtTrace.Info("TestSourceHost: Creating assembly resolver with resolution paths {0}.", string.Join(",", resolutionPaths.ToArray()));
+                }
+
+                // Adding adapter folder to resolution paths
+                if (!resolutionPaths.Contains(Path.GetDirectoryName(typeof(TestSourceHost).Assembly.Location)))
+                {
+                    resolutionPaths.Add(Path.GetDirectoryName(typeof(TestSourceHost).Assembly.Location));
+                }
+
+                // Adding extensions folder to resolution paths
+                if (!resolutionPaths.Contains(Path.GetDirectoryName(typeof(AssemblyHelper).Assembly.Location)))
+                {
+                    resolutionPaths.Add(Path.GetDirectoryName(typeof(AssemblyHelper).Assembly.Location));
+                }
+            }
+
+            // Case when DisableAppDomain setting is present in runsettings and no child-appdomain is created
+            if (this.runSettings != null && MSTestAdapterSettings.IsAppDomainCreationDisabled(this.runSettings.SettingsXml))
```

Comment:
>this.runSettings != null && MSTestAdapterSettings.IsAppDomainCreationDisabled(this.runSettings.SettingsXml) [](start = 16, length = 107)

nitpick: might want to have a single function and replace the usages..

Reply:
Done.